src/Orleans.Core/Configuration/OptionLogger/DefaultOptionsFormatter.cs

```diff
@@ -30,21 +32,60 @@ public IEnumerable<string> Format()
             return typeof(T)
                 .GetProperties()
                 .Where(prop => prop.GetGetMethod() != null && prop.GetSetMethod() != null)
-                .Select(FormatProperty)
+                .SelectMany(FormatProperty)
                 .ToList();
         }
 
-        private string FormatProperty(PropertyInfo property)
+        private IEnumerable<string> FormatProperty(PropertyInfo property)
         {
+            var result = new List<string>();
             var name = property.Name;
             var value = property.GetValue(this.options);
             var redactAttribute = property.GetCustomAttribute<RedactAttribute>(inherit: true);
 
-            return OptionFormattingUtilities.Format(
-                name, 
-                redactAttribute != null 
-                    ? redactAttribute.Redact(value)
-                    : value);
+            // If it is a dictionary -> one line per item
+            if (typeof(IDictionary).IsInstanceOfType(value))
+            {
+                var dict = (IDictionary)value;
+                foreach (DictionaryEntry kvp in dict)
+                {
+                    result.Add(OptionFormattingUtilities.Format($"{nameof(dict)}.{kvp.Key}", RedactIfNeeded(redactAttribute, kvp.Value)));
```

Comment:
The fixed string "dict" is a little weird here. Maybe `$"{name}[{kvp.Key}]"` instead?
Similarly for ICollection, we could keep each value on a separate line and format it as `$"{name}[{index}]"` -- I'm either way about this

Reply:
Yep wrong copy paste, should be `name`. I like your proposal for `ICollection`